connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -432,7 +441,9 @@ private void openPartitions(Collection<TopicPartition> partitions) {
     }
 
     private void closePartitions() {
-        commitOffsets(time.milliseconds(), true);
+        if (!context.offsetCommitDisabled()) {
+            commitOffsets(time.milliseconds(), true);
```

Comment:
This closes the task since the second parameter is true -- I don't think we can just skip over this step entirely. I'm also not sure what we expect the semantics for flush() to be if offset commit is not enabled (and this might need clarification in the KIP as well). The flush() call here may not be superfluous if the connector uses that as a signal that it should flush and commit offsets (all in the destination system).